_posts/current/blog/2022-01-06-how-to-emulate-trusted-platform-module-in-qemu-with-u-boot.md

```diff
@@ -0,0 +1,101 @@
+
+## Building U-Boot
+
+{% include image.html path="/assets/images/content/building-uboot-image-1.png" alt="Building Uboot image 1" %}
+
+## Running QEMU
+
+Make sure swtpm is installed and running on your system. For Debian and friends there's a swtpm package, so just do
+
+{% include image.html path="/assets/images/content/running-qemu-image-1.png" alt="Running EQMU image 1" %}
+
+and launch QEMU with swtpm support
+
+{% include image.html path="/assets/images/content/running-qemu-image-2.png" alt="Running EQMU image 2" %}
+
+## Booting linux
+
+From U-Boot's command line do something along the lines of
+
+{% include image.html path="/assets/images/content/booting-linux-image-1.png" alt="Booting Linux image 1" %}
+
+If everything is compiled and launched correctly, you should see the kernel reporting the location of some related EventLog pointers.
+
+{% include image.html path="/assets/images/content/booting-linux-image-2.png" alt="Booting Linux image 2" %}
+
+## Reading the EventLog
+
+I am using a debian qcow2 image, where I have installed the latest tpm2 tools. If you don't have them install them with
+
+{% include image.html path="/assets/images/content/reading-the-eventlog-image-1" alt="Reading the eventlog image 1" %}
+
+The kernel exposes the eventlog in /sys. So you can read it with:
+
+{% include image.html path="/assets/images/content/reading-the-eventlog-image-2" alt="Reading the eventlog image 2" %}
+
+{% include image.html path="/assets/images/content/reading-the-eventlog-image-3" alt="Reading the eventlog image 3" %}
+
+{% include image.html path="/assets/images/content/reading-the-eventlog-image-4" alt="Reading the eventlog image 4" %}
+
+{% include image.html path="/assets/images/content/reading-the-eventlog-image-5" alt="Reading the eventlog image 5" %}
+
+{% include image.html path="/assets/images/content/reading-the-eventlog-image-6" alt="Reading the eventlog image 6" %}
+
+{% include image.html path="/assets/images/content/reading-the-eventlog-image-7" alt="Reading the eventlog image 7" %}
+
+{% include image.html path="/assets/images/content/reading-the-eventlog-image-8" alt="Reading the eventlog image 8" %}
+
+{% include image.html path="/assets/images/content/reading-the-eventlog-image-9" alt="Reading the eventlog image 9" %}
+
+{% include image.html path="/assets/images/content/reading-the-eventlog-image-10" alt="Reading the eventlog image 10" %}
+
+{% include image.html path="/assets/images/content/reading-the-eventlog-image-11" alt="Reading the eventlog image 11" %}
+
+{% include image.html path="/assets/images/content/reading-the-eventlog-image-12" alt="Reading the eventlog image 12" %}
+
+{% include image.html path="/assets/images/content/reading-the-eventlog-image-13" alt="Reading the eventlog image 13" %}
+
```

Comment:
@EbbaSimpson These are missing png/jpg too

Reply:
I know I just realised.. doh!! Hopefully I have fixed this now.